src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1560,9 +1556,17 @@ private TypeWithState VisitRefExpression(BoundExpression expr, TypeWithAnnotatio
 
         private bool TryGetReturnType(out TypeWithAnnotations type)
         {
-            var method = (MethodSymbol)_symbol;
-            var returnType = (_useMethodSignatureReturnType ? _methodSignatureOpt : method).ReturnTypeWithAnnotations;
-            Debug.Assert((object)returnType != LambdaSymbol.ReturnTypeIsBeingInferred);
+            var method = _symbol as MethodSymbol;
+            var nullableReturnType = (_methodSignatureOpt ?? method)?.ReturnTypeWithAnnotations;
+            Debug.Assert(!nullableReturnType.HasValue || (object)nullableReturnType.Value != LambdaSymbol.ReturnTypeIsBeingInferred);
+
+            if (!nullableReturnType.HasValue)
```

Comment:
>nullableReturnType [](start = 17, length = 18)

The `TypeWithAnnotations?` variable seems unnecessary. Consider simply returning early if the method is `null`.
```C#
var method = _methodSignatureOpt ?? (_symbol as MethodSymbol);
if (method is null)
{
    type = default;
    return false;
}

var returnType = method.ReturnTypeWithAnnotations;
```